### ISSUE TITLE
Rom Cache System -- Generalized Database.

### ISSUE THREAD

**Issue by GoogleCodeExporter** [jennorby/mupen64plus#75]
```
From Issue 11:

We need to work out a system where all areas of mupen64plus can have access
to cached rom information, not just the rom browser. That way, the cheat
system can be tied into it also, so it won't have to maintain its own Rom
info (CRC, name, etc).
```

Original issue reported on code.google.com by `ebenbl...@gmail.com` on 22 Apr 2008 at 9:27

**Comment by GoogleCodeExporter** [comment]
```
I agree, issues 77, and 47 would also benefit from moving the rom caching 
information from the GUI and putting it in core where all the plugins could 
access. 
Since MD5 is suggested as the mechanism for assigning GoodNames, issue 66 would 
get 
fixed also. 

Just a reference, here are all the cached options from Project64:

Goodname
Country
Size
Comments / Notes (User)
Filename

Catride ID
CRC1
CRC2
CIC Chip
Developer
Filename
Force Feedback
Genre
Internal Name
Notes (Core)
Notes (Plugins)
Players
Release Date

I'm not suggesting to implement all of these. The Developer, and Genre seem 
somewhat silly to me and we don't want to maintain too large a database of 
items 
not intrinsically in the ROM itself. We also might want to use MD5 instead of 
dual 
CRC to assign GoodNames and hence cheats.
```

Original comment by `sknau...@wesleyan.edu` on 23 Apr 2008 at 7:42

**Comment by GoogleCodeExporter** [comment]
```
I wrote up this document to aid us in the specifications of this system. This is
actually a high priority because a lot of things depend on it.
```

Original comment by `sgorman07@gmail.com` on 8 May 2008 at 1:25
- Changed title: **Rom Cache System -- Generalized Database.**
- Added labels: **Priority-High**
- Removed labels: **Priority-Medium**

Attachments:
- [romcachesystem.pdf](https://storage.googleapis.com/google-code-attachments/mupen64plus/issue-75/comment-2/romcachesystem.pdf)

**Comment by GoogleCodeExporter** [comment]
```
After the branch is created you should put a version of this document in ODF or
whatever modifiable format in the doc folder, so that it can be kept up to date.
```

Original comment by `richard...@gmail.com` on 8 May 2008 at 1:51

**Comment by GoogleCodeExporter** [comment]
```
I made okaygo's document into a wiki page, rather than keep it in svn:

http://code.google.com/p/mupen64plus/wiki/RomCacheSystem
```

Original comment by `ebenbl...@gmail.com` on 8 May 2008 at 8:32

**Comment by GoogleCodeExporter** [comment]
```
I will be working on this branch... feel free to help out how you see fit!
```

Original comment by `sgorman07@gmail.com` on 8 May 2008 at 10:09

**Comment by GoogleCodeExporter** [comment]
```
This needs to be done ASAP
```

Original comment by `sgorman07@gmail.com` on 11 May 2008 at 8:19
- Added labels: **Milestone-1.4**, **Priority-Critical**
- Removed labels: **Priority-High**

**Comment by GoogleCodeExporter** [comment]
```
Critical priority is for bugs which are breaking builds or causing severe
malfunctions with normal program usage.  This is really a new feature which 
will fix
a couple of problems.  Normally it would be Medium but since you indicated that 
you
want to finish it before 1.4, it should be High.
```

Original comment by `richard...@gmail.com` on 11 May 2008 at 8:37
- Added labels: **Priority-High**
- Removed labels: **Priority-Critical**

**Comment by GoogleCodeExporter** [comment]
```
I'd say with commit 554 this is Fixed. :) Richard wants to get 1.4 out sooner 
rather than later, so the merge to trunk will occur right after the release. I 
personally disagree, but don't think its important enough to argue. Setting to 
fixed and removing 1.4 milestone.
```

Original comment by `sknau...@wesleyan.edu` on 13 Jun 2008 at 5:11
- Changed state: **Fixed**
- Removed labels: **Milestone-1.4**
